src/main/java/com/github/jreddit/oauth/client/RedditHttpClient.java

```diff
@@ -0,0 +1,143 @@
+package com.github.jreddit.oauth.client;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.jreddit.oauth.RedditToken;
+import com.github.jreddit.request.RedditGetRequest;
+import com.github.jreddit.request.RedditPostRequest;
+
+/**
+ * HTTP client implementation for a <i>RedditClient</i>.
+ * 
+ * @author Simon Kassing
+ * 
+ * @see RedditClient
+ */
+public class RedditHttpClient extends RedditClient {
+
+	/** Logger for this class. */
+	final static Logger LOGGER = LoggerFactory.getLogger(RedditHttpClient.class);
+	
+	/** User agent. */
```

Comment:
comments like "User agent", "Constructor" are not very helpful neither for the developer looking at the codebase, nor the user using the library. I suggest removing them, because it's just cluttering the codebase.